Sources/MessagingInApp/Views/GistInlineInAppMessageView.swift

```diff
@@ -15,20 +15,255 @@ public protocol GistInlineMessageUIViewDelegate: AnyObject {
+
+    // Determines if the View is already trying to show a message or not.
+    private var isRenderingOrDisplayingAMessage: Bool {
+        inlineMessageManager != nil
+    }
+
+    // Subscriber to listen to changes in InAppMessageState.
+    private var inAppMessageStoreSubscriber: InAppMessageStoreSubscriber?
+
     public init(elementId: String) {
         super.init(frame: .zero)
         self.elementId = elementId
+
+        // Setup the View and display a message, if one available. Since an elementId has been set.
+        setupView()
+        refreshView()
     }
 
     // This is called when the View is created from a Storyboard.
     required init?(coder: NSCoder) {
         super.init(coder: coder)
+
+        setupView()
+        // An element id will not be set yet. No need to check for messages to display.
+    }
+
+    deinit {
+        unsubscribeFromInAppMessageState()
+    }
+
+    private func setupView() {
+        subscribeToInAppMessageState()
+    }
+
+    private func subscribeToInAppMessageState() {
+        // Begin listening to the queue for new messages.
+        inAppMessageStoreSubscriber = {
+            let subscriber = InAppMessageStoreSubscriber { [weak self] state in
+                // InAppMessageManager callback function might not be on UI thread.
+                // Switch to UI thread to update UI.
+                Task { @MainActor [weak self] in
+                    self?.refreshView(state: state)
+                }
+            }
+            // Subscribe to changes in `currentMessageState` property of `InAppMessageState`
+            inAppMessageManager.subscribe(keyPath: \.currentMessageState, subscriber: subscriber)
+            return subscriber
+        }()
+    }
+
+    private func unsubscribeFromInAppMessageState() {
+        guard let subscriber = inAppMessageStoreSubscriber else { return }
+
+        logger.logWithModuleTag("Unsubscribing GistInlineMessageUIView from InAppMessageState", level: .debug)
+        inAppMessageManager.unsubscribe(subscriber: subscriber)
+        inAppMessageStoreSubscriber = nil
+    }
+
+    private func refreshView(forceShowNextMessage: Bool = false) {
+        inAppMessageManager.fetchState { [self] state in
+            refreshView(state: state, forceShowNextMessage: forceShowNextMessage)
+        }
+    }
+
+    // Updates the state of the View, if needed. Call as often as you need if an event happens that may cause the View to need to update.
+    private func refreshView(state: InAppMessageState, forceShowNextMessage: Bool = false) {
+        defer {
+            // Always call the refreshViewListener at the end of the function to know processing is done.
+            refreshViewListener?()
+        }
+
+        guard let elementId = elementId else {
+            return // we cannot check if a message is available until element id set on View.
+        }
+
+        let queueOfMessagesForGivenElementId = state.inlineMessagesInQueue(forElementId: elementId)
```

Comment:
this is interesting we might need to keep queue of embedded messages, indeed but for now lets start off with one embedded message?

Reply:
Updated